### PR TITLE
txr chapter 15

### PR DESCRIPTION
**Pull request by txr12**

### CODE REVIEW

File: chapter15/txr/ex1.pl [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/perl -w
+use 5.14.0;
+
+my $secret_number = get_random();
+my $exit;
+
+say 'enter a integer between 1 to 100 (included) ';
+say 'enter \'exit\' or \'quit\' or simply press enter to exit ';
+until ($exit) {
```
Comment (a-panayotopoulos) [inline chapter15/txr/ex1.pl]: This doesn't seem to be used; could use `while( 1 )` instead.

File: chapter15/txr/ex3.pl [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		continue;
+	}
+	default {
+
+		$permissions =  $permissions // 'unknown';
+		$permissions =~ s/,$//;
+		say "$_ is  $permissions" ;
+		$permissions = undef;
+	}
+}
```
Comment (a-panayotopoulos) [inline chapter15/txr/ex3.pl]: Nice solution.

File: chapter15/txr/ex4.pl [hunk 3]
```diff
@@ -0,0 +1,29 @@
+use 5.014;
+use Scalar::Util qw /looks_like_number/;
+
+my $number = $ARGV[0];
+
+given (looks_like_number($number)) {
+	when ( ~~ 0 ) {
```
Comment (a-panayotopoulos) [inline chapter15/txr/ex4.pl]: Granted, the exercise specifically asked for smart matching, so I can't really fault you on this, but you've got to admit that this is pretty horrible.  8^]

File: chapter15/txr/ex1.pl (hunk 1 above)
Comment (txr12) [inline chapter15/txr/ex1.pl]: lol I wonder if you made the same comment back on chapter 10:)

File: chapter15/txr/ex4.pl (hunk 3 above)
Comment (txr12) [inline chapter15/txr/ex4.pl]: Yeah it is. I need to look at the book solution as I am curious to what they did. I don't mind so much the use in ex5.pl 7 ~~ @divisors but the ~~ 0 is awful

Comment (a-panayotopoulos) [inline chapter15/txr/ex4.pl]: Don't look. It's even worse.  8^(

**Pull request merged**